alerting/src/main/kotlin/com/amazon/opendistroforelasticsearch/alerting/TriggerService.kt

```diff
@@ -57,11 +61,48 @@ class TriggerService(val client: Client, val scriptService: ScriptService) {
     }
 
     // TODO: This is a placeholder to write MonitorRunner logic, it can be replaced with the actual implementation when available
+    @Suppress("UNCHECKED_CAST")
     fun runAggregationTrigger(
         monitor: Monitor,
         trigger: AggregationTrigger,
         ctx: AggregationTriggerExecutionContext
     ): AggregationTriggerRunResult {
-        return AggregationTriggerRunResult(triggerName = trigger.name, error = null, aggregationResultBuckets = mapOf())
+        return try {
+            val bucketIndices =
+                ((ctx.results[0]["aggregations"] as HashMap<*, *>)[trigger.id] as HashMap<*, *>)["bucket_indices"] as List<*>
+            val parentBucketPath =
+                ((ctx.results[0]["aggregations"] as HashMap<*, *>)[trigger.id] as HashMap<*, *>)["parent_bucket_path"] as String
+            val aggregationPath = AggregationPath.parse(parentBucketPath)
+            // TODO test this part by passing sub-aggregation path
+            var parentAgg = (ctx.results[0].get("aggregations") as HashMap<*, *>)
+            aggregationPath.pathElementsAsStringList.forEach { sub_agg ->
+                parentAgg = (parentAgg[sub_agg] as HashMap<*, *>)
+            }
+            val buckets = parentAgg["buckets"] as List<*>
+            val selectedBuckets = mutableMapOf<String, AggregationResultBucket>()
+            for (bucketIndex in bucketIndices) {
+                val bucketDict = buckets[bucketIndex as Int] as Map<String, Any>
+                val bucketKeyValuesList = getBucketKeyValuesList(bucketDict)
+                val aggResultBucket = AggregationResultBucket(parentBucketPath, bucketKeyValuesList, bucketDict)
+                selectedBuckets[aggResultBucket.getBucketKeysHash()] = aggResultBucket
+            }
+            AggregationTriggerRunResult(trigger.name, null, selectedBuckets)
+        } catch (e: Exception) {
+            logger.info("Error running script for monitor ${monitor.id}, trigger: ${trigger.id}", e)
+            // TODO empty map here with error should be treated in the same way as TraditionTrigger with error running script
+            AggregationTriggerRunResult(trigger.name, e, emptyMap())
+        }
+    }
+
+    @Suppress("UNCHECKED_CAST")
+    private fun getBucketKeyValuesList(bucket: Map<String, Any>): List<String> {
+        val keyValuesList = mutableListOf<String>()
+        when {
+            bucket["key"] is String -> keyValuesList.add(bucket["key"] as String)
+            bucket["key"] is Map<*, *> -> (bucket["key"] as Map<String, Any>).values.map { keyValuesList.add(it as String) }
```

Comment:
Can we check for the case what it prints when a bucket value is a multi-bucket aggregation. Also, in this case what should be the expected behavior as we may not want to print the sub-aggregations.

Reply:
As we discussed offline, this isn't related since this is only accessing the bucket key values